posts/2023-06-01-Rust-1.70.0.md

```diff
@@ -0,0 +1,74 @@
+
+### `OnceCell` and `OnceLock`
+
+Two new types have been stabilized for one-time initialization of shared data, `OnceCell` and its thread-safe counterpart `OnceLock`. These can be used anywhere that immediate construction is not wanted, and perhaps not even possible like non-`const` data in global variables.
+
+```rust
+use std::sync::OnceLock;
+
+fn main() {
+    let winner_lock = OnceLock::new();
+
+    let winner = std::thread::scope(|s| {
+        s.spawn(|| winner_lock.set("thread"));
+
+        std::thread::yield_now(); // give them a chance...
+
+        winner_lock.get_or_init(|| "main")
+    });
+
+    println!("{winner} wins!");
+}
+```
+
+Crates such as `lazy_static` and `once_cell` have filled this need in the past, but now these building blocks are part of the standard library, ported from `once_cell`'s `unsync` and `sync` modules. There are still more methods that may be stabilized in the future, as well as companion `LazyCell` and `LazyLock` types that store their initializing function, but this first step in stabilization should already cover many use cases.
+
+### Named levels of debug information
+
+The `-Cdebuginfo` compiler option has previously only supported numbers 0..=2 for increasing amounts of debugging information, where Cargo defaults to 2 in dev and test profiles and 0 in release and bench profiles. These debug levels can now be set by name: "none" (0), "limited" (1), and "full" (2), as well as two new levels, "line-directives-only" and "line-tables-only".
+
+The Cargo and rustc documentation both called level 1 "line tables only" before, but it was more than that with information about all functions, just not types and variables. That level is now called "limited", and the new "line-tables-only" level is further reduced to the minimum needed for backtraces with filenames and line numbers. This may eventually become the level used for `-Cdebuginfo=1`. The other `line-directives-only` level is intended for NVPTX profiling, and is otherwise not recommended.
+
+### Enforced stability in the `test` CLI
+
+When `#[test]` functions are compiled, the executable gets a command-line interface from the `test` crate. This CLI has a number of options, including some that are not yet stabilized and require specifying `-Zunstable-options` as well, like many other commands in the Rust toolchain. However, while that's only intended to be allowed in nightly builds, that restriction wasn't active in `test` -- until now. Starting with 1.70.0, stable and beta builds of Rust will no longer allow unstable `test` options, making them truly nightly-only as documented.
+
+### Stabilized APIs
+
+- [`thing`](link)
```

Comment:
Do you need help with getting these generated?

Reply:
I think they're in the latest release notes (so just needs copying)